addons/account_edi_ubl/models/account_edi_format.py

```diff
@@ -159,3 +157,85 @@ def _find_value(xpath, element=tree):
                             invoice_line_form.tax_ids.add(tax)
 
         return invoice_form.save()
+
+    ####################################################
+    # Export
+    ####################################################
+
+    def _get_ubl_values(self, invoice):
+        ''' Get the necessary values to generate the XML. These values will be used in the qweb template when
+        rendering. Needed values differ depending on the implementation of the UBL, as (sub)template can be overriden
+        or called dynamically.
+        :returns:   a dictionary with the value used in the template has key and the value as value.
+        '''
+        def format_monetary(amount):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.85000000000001).
+            return float_repr(amount, invoice.currency_id.decimal_places)
+
+        return {
+            **invoice._prepare_edi_vals_to_export(),
+            'ubl_version': 2.1,
+            'type_code': 380 if invoice.move_type == 'out_invoice' else 381,
+            'payment_means_code': 42 if invoice.journal_id.bank_account_id else 31,
+            'bank_account': invoice.partner_bank_id,
+            'format_monetary': format_monetary,
+            'customer_vals': {'partner': invoice.commercial_partner_id},
+            'supplier_vals': {'partner': invoice.company_id.partner_id.commercial_partner_id},
```

Comment:
This is confusing in so many ways.  supplier_vals should be company_vals or something.  (or emittor_vals e.g.)

Reply:
I think company/partner is more confusing than customer/supplier. If we want to support bills and invoices, the use of company/partner is so confusing when you need to match them with odoo's fields. Customer/supplier is the name defined by BIS3 (see the xml), so I think this is as clear as can be.

Reply:
Ok, I did not see they used that term in the XML